charts/shared/common-gitops/templates/crossplane/_awsblockPublicBuckets.tpl

```diff
@@ -0,0 +1,29 @@
+{{/* vim: set filetype=mustache: */}}
+{{/*
+Render the restrictPublicBuckets, blockPublicAcls, blockPublicPolicy value for AWS resource:
+Accepts dict:
+{
+  root: [map] - root context
+  kind: [string] - resource kind name, e.g. "BucketPublicAccessBlock"
+  name: [string] - item id, e.g. "<bucket_alias>"
+}
+Sample return boolean:
+true
+
+Global Override:
+awsBlockBucketPublic: false
+
+*/}}
+{{- define "common-gitops.crossplane.blockPublicBuckets" -}}
+  {{- $kindObj := (index .root.Values .kind) -}}
+  {{- $item := (index $kindObj.items .name) -}}
+  {{- /* Take the first non-empty argument */ -}}
+  {{- coalesce ($item.forProvider).restrictPublicBuckets
```

Comment:
Wouldn't it be confusing for the end-user to be able to set one variable and get it's value in another one? I.e. if user sets blockPublicPolicy it's value will be used for all 3 parameters.

Reply:
So, for the sake of clarity, it might make sense to either have 3 different templates, but in this case each template will be used only once what defeats the purpose of template, or put `coalesce ...` inline into `bucket-public-access-block.yaml`.

Reply:
Hi,
My initial understanding was to enforce blocking of public access entirely by setting value of all three parameter to true.So even if user sets blockPublicAcls and blockPublicPolicy: false it would be mutated to true. If required override all 3 at once with global value.

I have reverted to only one parameter i.e restrictPublicBuckets. This blocks all public access and if user wants to set granular access with blockPublicAcls and blockPublicPolicy, it will allow the user to do so.

Ideally, upbound provider should have defaulted these values to True so we would not need this code change at our end.

Reply:
Ignore my pervious comment. restrictPublicBuckets is also a granular level permission. It does not block all public access my bad.

Reply:
@gmykhailiuta  Please review. I parameterized the helper function _awsblockPublicBuckets.tpl

But there is one help I need. Not sure why it's returning string instead of boolean. I tried without quotes but yet same result.